### ISSUE TITLE
0000068:
Implement caching of data

### ISSUE THREAD

**Issue by Gloirin** [tine20/Tine-2.0-Open-Source-Groupware-and-CRM#129]
**Reported by pschuele on 11 Jul 2008 14:34**

**Version:** 2008-summer-ms1

Implement caching of data

- use Zend_Cache
- profile with xdebug to find functions/data which are performance killers and cache the data
- work on the milestone1 branch
- configure cache with config.ini
- support memcache server and file caching
- clean the cache if data gets updated (for example container grants)

**Comment by Gloirin** [comment]
Related to #144

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 18 Jul 2008 12:32**

profile addressbook -&gt; why is it taking so long to load? should be about 200 ms and is 1 second (with the demo data)

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 18 Jul 2008 13:03**

removed account_grants from Addressbook/Json.php to improve the performance by 50% -&gt; see related task 100.

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 21 Jul 2008 10:01**

- make Zend_Cache available via Zend_Registry (if it is active in config.ini)
- added first caching to function:
Tinebase_Container: getContainerById()

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 21 Jul 2008 13:47**

- added caching for getRelations (Tinebase_Relations) and getContainerByACL (Tinebase_Container)
- merged changes to trunk/tine20
- added config.ini options

still todo: 
- add memcache support
- find more data that should be cached

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 21 Jul 2008 15:52**

- added more possible caching options to config.ini
- memcache is working now as well (but it doen&#039;t support tags, thats bad for getContainerByACL-caching).
- cache more data - but i think that would be a new task (lots of profiling should be done first). closing this one ....
